### DISCUSSION TITLE
Passing CorrelationId with distributed Eto?

### DISCUSSION THREAD

**Discussion by piotr07da** [abpframework/abp#9769] [Q&A]
Hi,

Is there a built in way of passing a metadata like CorrelationId (or Timestamp or anything additional) with the distributed events?

It is essential for creating an integration patterns like a process managers.

One solution I can think of is to create my own class
`public class EtoEnvelope<T>
{
    public T Event { get; set; }
    public Guid CorrelationId { get; set; }
}`
to wrap up all the additional stuff I need.

I wonder if creating such class is a good idea or there is something built in that I should use instead?

Thanks

**Comment by maliming** [comment]
I think you need to set it manually.
